cirq-ft/cirq_ft/algos/arithmetic_gates.py

```diff
@@ -161,9 +222,104 @@ def __pow__(self, power: int):
     def __repr__(self) -> str:
         return f'cirq_ft.LessThanEqualGate({self.x_bitsize}, {self.y_bitsize})'
 
+    def _decompose_via_tree(
+        self, context: cirq.DecompositionContext, X: Tuple[cirq.Qid, ...], Y: Tuple[cirq.Qid, ...]
+    ) -> cirq.OP_TREE:
+        """Returns comparison oracle from https://www.nature.com/articles/s41534-018-0071-5#Sec8"""
+        if len(X) == 1:
+            return
+        if len(X) == 2:
+            yield mix_double_qubit_registers(context, (X[0], Y[0]), (X[1], Y[1]))
+            return
+
+        m = len(X) // 2
+        yield self._decompose_via_tree(context, X[:m], Y[:m])
+        yield self._decompose_via_tree(context, X[m:], Y[m:])
+        yield mix_double_qubit_registers(context, (X[m - 1], Y[m - 1]), (X[-1], Y[-1]))
+
+    def _decompose_with_context_(
+        self, qubits: Sequence[cirq.Qid], context: Optional[cirq.DecompositionContext] = None
+    ) -> cirq.OP_TREE:
+        if context is None:
+            context = cirq.DecompositionContext(cirq.ops.SimpleQubitManager())
+
+        P, Q, target = (qubits[: self.x_bitsize], qubits[self.x_bitsize : -1], qubits[-1])
+
+        n = min(len(P), len(Q))
+
+        prefix_equality = None
+        adjoint: List[cirq.Operation] = []
+
+        # if one of the registers is longer than the other store equality with |0--0>
+        # into `prefix_equality` using d = |len(P) - len(Q)| And operations => 4d T.
+        if len(P) != len(Q):
+            [prefix_equality] = context.qubit_manager.qalloc(1)
```

Comment:
I think we usually do (x,) to do one-thing-unpacking

Reply:
done